File: langs/fortran/prime.f90
program prime
  implicit none
  integer :: count, length, n
  character(:), allocatable :: arg

  count = command_argument_count()
  n = 10000000
  if (count >= 1) then
     call get_command_argument(1, length = length)
     allocate (character(length) :: arg)
     call get_command_argument(1, arg)
     read (arg, *, err=100) n
     if ( n < 0 ) goto 100
     deallocate (arg)
  end if
  print '(i0)', count_prime(n)
  stop

100 stop 1

contains
  function sieve(size)
    integer, intent(in) :: size
    logical :: sieve(size)
    integer :: root, i

    sieve = .true.
    if (size == 0) then
       return
    end if
    sieve(1) = .false.
    root = int(sqrt(dble(size)) + 1)
    do i = 1, root
       if (sieve(i)) then
          sieve(i * i : size : i) = .false.
       end if
    end do
  end function sieve

  function count_prime(n)
    integer, intent(in) :: n
    integer :: count_prime, i
    logical :: sieve_array(n)

    count_prime = 0
    sieve_array = sieve(n)
    do i = 1, n
       if (sieve_array(i)) then
          count_prime = count_prime + 1
       end if
    end do
  end function count_prime
end program prime

File: langs/c++/prime.cc
#include <algorithm>
#include <iostream>
#include <iterator>
#include <string>
#include <vector>
#include <cmath>

std::vector<bool> sieve(int size) {
    std::vector<bool> sieveArray(size, true);
    for (int i = 0; i < std::min(2, size); ++i) { sieveArray[i] = false; }
    int root = std::sqrt(size) + 1;
    for (int i = 2; i < root; ++i) {
        if (!sieveArray[i]) { continue; }
        for (int j = i * i; j < size; j += i) {
            sieveArray[j] = false;
        }
    }
    return sieveArray;
}

int countPrime(int n) {
    std::vector<bool> sieveArray = sieve(n + 1);
    return std::count(sieveArray.begin(), sieveArray.end(), true);
}

int main(int argc, char* argv[]) {
    int n = (argc >= 2) ? std::stoi(argv[1]) : 10000000;
    std::cout << countPrime(n) << std::endl;
}
